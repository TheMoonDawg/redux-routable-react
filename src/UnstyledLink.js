import Link from './Link'
import PropTypes from 'prop-types'
import React from 'react'

const linkStyle = {
  color: 'initial',
  cursor: 'initial',
  textDecoration: 'initial',
}

const UnstyledAnchor = ({ style, ...props }) => (
  <a {...props} style={{ ...linkStyle, ...style }} />
)

UnstyledAnchor.propTypes = {
  style: PropTypes.object,
}

const UnstyledLink = props => <Link {...props} component={UnstyledAnchor} />

export default UnstyledLink
