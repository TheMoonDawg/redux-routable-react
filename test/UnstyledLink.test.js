import UnstyledLink from '../src/UnstyledLink'
import testLink from './testLink'

testLink('UnstyledLink', UnstyledLink)
