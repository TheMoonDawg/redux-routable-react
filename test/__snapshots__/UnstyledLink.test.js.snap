// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnstyledLink links to Home Page 1`] = `
<a
  href="/"
  onClick={[Function]}
  style={
    Object {
      "color": "initial",
      "cursor": "initial",
      "textDecoration": "initial",
    }
  }
>
  Home
</a>
`;

exports[`UnstyledLink links to User Page with required path param 1`] = `
<a
  href="/user/123"
  onClick={[Function]}
  style={
    Object {
      "color": "initial",
      "cursor": "initial",
      "textDecoration": "initial",
    }
  }
>
  User 123
</a>
`;

exports[`UnstyledLink links to Users Page with optional path and query params 1`] = `
<a
  href="/users/admin?name_query=john"
  onClick={[Function]}
  style={
    Object {
      "color": "initial",
      "cursor": "initial",
      "textDecoration": "initial",
    }
  }
>
  All Users
</a>
`;

exports[`UnstyledLink links to Users Page with optional path param 1`] = `
<a
  href="/users/admin"
  onClick={[Function]}
  style={
    Object {
      "color": "initial",
      "cursor": "initial",
      "textDecoration": "initial",
    }
  }
>
  All Users
</a>
`;

exports[`UnstyledLink links to Users Page with query param 1`] = `
<a
  href="/users?name_query=john"
  onClick={[Function]}
  style={
    Object {
      "color": "initial",
      "cursor": "initial",
      "textDecoration": "initial",
    }
  }
>
  All Users
</a>
`;
